pkg/hwinfo/hwinfo.go

```diff
@@ -1,16 +1,48 @@
 package hwinfo
 
 import (
-	"errors"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
 )
 
-var ErrNotPresent = errors.New("hwinfo: not present")
+type pciDeviceInfo struct {
+	Address     string `json:"address"`
+	DeviceType  string `json:"device_type,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	ProductName string `json:"product_name"`
+	Description string `json:"description,omitempty"`
+}
+
+type usbDeviceInfo struct {
+	Address     string `json:"address,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	DeviceID    string `json:"id"`
+	Description string `json:"description,omitempty"`
+}
+
+type monitorInfo struct {
+	ID          string `json:"id"`
+	Description string `json:"description,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	Size        string `json:"size,omitempty"`
+	Resolution  string `json:"resolution,omitempty"`
+}
 
 func Inventory() (map[string]interface{}, error) {
 	hw, err := fetchInventory()
 	if err != nil {
-		return nil, err
+		err = errors.Wrap(err, "hwinfo")
+		log.Error(err)
+		return hw, err
 	}
 
 	return hw, nil
 }
+
+func mergeStringMaps(mapA, mapB *map[string]interface{}) map[string]interface{} {
```

Comment:
In Go it's common to have simple private helpers, this thing contains a simple `for {}` loop, why move it to another package, also its effect can be seen here — what exactly it does. Otherwise it would require a package lookup.
See [Proverb No.8](https://go-proverbs.github.io)

Reply:
I just expressed my opinion. I don't disagree to have simple private helpers. I _personally_ think that this helper is so generic, that there is no need to have it as a _private_ function inside this package.
I agree that some copying is often better than adding an _external_ dependency but this lies entirely within our codebase.
Different opinions welcome :)

Reply:
@soupdiver sorry if my comment seemed harsh, from my history of legacy codebases, moving primitive functions into packages like "common" was never a good decision - it still get used only once, and there can be multiple packages called "util" and "common" in the end, with functions like left-pad or isArray (node.js pun intended). So we've got different views on this, haha

Reply:
Fair reasoning
I can live with that argumentation :)

> sorry if my comment seemed harsh

all good 😁